plugins/obs-filters/noise-suppress-filter.c

```diff
@@ -23,19 +34,30 @@
 /* -------------------------------------------------------- */
 
 #define S_SUPPRESS_LEVEL "suppress_level"
+#define S_USE_RNNOISE "use_rnnoise"
 
 #define MT_ obs_module_text
 #define TEXT_SUPPRESS_LEVEL MT_("NoiseSuppress.SuppressLevel")
+#define TEXT_USE_RNNOISE MT_("NoiseSuppress.UseRnnoise")
+#define TEXT_USE_RNNOISE_TT MT_("NoiseSuppress.UseRnnoise.TT")
 
 #define MAX_PREPROC_CHANNELS 8
 
+/* RNNoise constants, these can’t be changed */
+#define RNNOISE_SAMPLE_RATE 48000
+#define RNNOISE_FRAME_SIZE 480
```

Comment:
Should these be configurable? Does it make sense to have this sample rate fixed to 48 kHz when I want to output a different sample rate?

Reply:
Should these just match OBS sample rate settings? Adding settings is a double-edged sword for usability if we can select reasonable defaults or set things automatically.

Reply:
As mentioned near those constants, these can’t be changed. The sample rate is the one used by the samples used to train the standard model of RNNoise, and the frame size is fixed in the library. Feeding non-resampled 44100Hz audio in the model will yield bad results.